curriculum/test/test-challenges.js

```diff
@@ -232,6 +232,29 @@ async function getChallenges(lang) {
         return [...challengeArray, ...flatten(challengesForBlock)];
       }, [])
   );
+  // This matches the order Gatsby uses (via a GraphQL query). Ideally both
+  // should be sourced and sorted using a single query, but we're not there yet.
+  challenges.sort((a, b) => {
```

Comment:
There's probably a nicer way of doing this, but it seems to do the job.

Reply:
If you want a cheaper way, I would go with sorting within the `reduce` above. But, sort is nice as is for the next person to read.

Just for brevity, using the `||` operator would make it look nicer:
```
if (a.superOrder < b.superOrder || a.order < b.order || a.challengeOrder < b.challengeOrder)
```
etc.

Reply:
@Sky020 The `||` operator can not be used in this case, because order of the properties in the sort is important.

I am putting together a little function that will work though and will suggest it for the line.

Reply:
I ended up googling and you can just return `(obj1.a - obj2.a || obj1.b - obj2.b || obj1.c - obj2.c)`, which I quite like.

Reply:
@RandellDawson , thank you, for letting me know. I cannot understand why, though?

Reply:
@Sky020 it's just a requirement of `Array.sort`. The comparison function has to return a number.  If it just returns a boolean, the array may not end up getting sorted.

Reply:
@Sky020 Actually, you could do as you were saying and Oliver's version is correct.  As long as all the properties are sorted in the same order, both your suggestion or @ojeytonwilliams' newer suggestion would work.  I was thinking of the use case where you want to sort multiple object properties where the direction of each property could be different.

@ojeytonwilliams Just as a sanity check, you could use the code in the following repl as the basis for a test to validate the sort works as intended.

https://repl.it/@rmdawson71/Sorting-objects-with-multiple-properties-by-ascending-order

Reply:
Hmm, I think I needed more coffee before replying - I probably misunderstood what you were getting at, @Sky020.  Either way, Randell's test looks good.  I will put that in.